src/main/java/org/opensearch/plugin/insights/rules/model/Measurement.java

```diff
@@ -0,0 +1,220 @@
+        }
+
+        if (total instanceof Long) {
+            return ((Long) total) / count;
+        } else if (total instanceof Integer) {
+            return ((Integer) total) / count;
+        } else if (total instanceof Double) {
+            return ((Double) total) / count;
+        } else if (total instanceof Float) {
+            return ((Float) total) / count;
+        } else {
+            throw new IllegalArgumentException("Unsupported number type: " + total.getClass());
+        }
+    }
+
+    /**
+     * Set measurement
+     * @param measurement measurement number
+     */
+    public void setMeasurement(Number measurement) {
+        number = measurement;
+    }
+
+    /**
+     * Set aggregation type
+     * @param aggregationType aggregation type
+     */
+    public void setAggregationType(AggregationType aggregationType) {
+        this.aggregationType = aggregationType;
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, ToXContent.Params params) throws IOException {
+        builder.startObject();
+        builder.field("number", number);
+        builder.field("count", count);
+        builder.field("aggregationType", aggregationType.toString());
+        builder.endObject();
+        return builder;
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        writeNumber(out, number);
+        out.writeInt(count);
+        out.writeString(aggregationType.toString());
+    }
+
+    private void writeNumber(StreamOutput out, Number number) throws IOException {
+        if (number instanceof Long) {
+            out.writeByte((byte) 0); // Type indicator for Long
+            out.writeLong((Long) number);
+        } else if (number instanceof Integer) {
+            out.writeByte((byte) 1); // Type indicator for Integer
+            out.writeInt((Integer) number);
+        } else if (number instanceof Double) {
+            out.writeByte((byte) 2); // Type indicator for Double
+            out.writeDouble((Double) number);
+        } else if (number instanceof Float) {
+            out.writeByte((byte) 3); // Type indicator for Float
+            out.writeFloat((Float) number);
+        } else {
+            throw new IOException("Unsupported number type: " + number.getClass());
+        }
+    }
+
+    private static Number readNumber(StreamInput in) throws IOException {
+        byte typeIndicator = in.readByte();
+        switch (typeIndicator) {
+            case 0:
+                return in.readLong();
+            case 1:
+                return in.readInt();
+            case 2:
+                return in.readDouble();
+            case 3:
+                return in.readFloat();
+            default:
+                throw new IOException("Unsupported number type indicator: " + typeIndicator);
+        }
+    }
+
+    public static Measurement readFromStream(StreamInput in) throws IOException {
+        Number number = readNumber(in);
```

Comment:
Why do we need to implement the `Writeable` interface? Probably can get rid of this if not really needed